packages/components/src/components/textarea/textarea-styles.ts

```diff
@@ -0,0 +1,146 @@
+const cssVariableResize = '--p-textarea-resize';
+
+export const getComponentCss = (
+  isDisabled: boolean,
+  hideLabel: BreakpointCustomizable<boolean>,
+  state: FormState,
+  hasCounter: boolean,
+  theme: Theme
+): string => {
+  const { primaryColor, contrastLowColor, contrastMediumColor, disabledColor } = getThemedColors(theme);
+  const {
+    primaryColor: primaryColorDark,
+    contrastLowColor: contrastLowColorDark,
+    contrastMediumColor: contrastMediumColorDark,
+    disabledColor: disabledColorDark,
+  } = getThemedColors('dark');
+  const { formStateColor, formStateHoverColor } = getThemedFormStateColors(theme, state);
+  const { formStateColor: formStateColorDark, formStateHoverColor: formStateHoverColorDark } = getThemedFormStateColors(
+    'dark',
+    state
+  );
+  return getCss({
+    '@global': {
+      ':host': {
+        display: 'block',
+        ...addImportantToEachRule({
+          ...colorSchemeStyles,
+          ...hostHiddenStyles,
+        }),
+      },
+      textarea: {
+        minHeight: `var(${cssVariableMinHeight}, 200px)`,
+        resize: `var(${cssVariableResize}, vertical)`,
+        display: 'block',
+        width: '100%',
+        height: 'auto',
+        margin: 0,
+        outline: 0,
+        WebkitAppearance: 'none', // iOS safari
+        appearance: 'none',
+        boxSizing: 'border-box',
+        border: `${borderWidthBase} solid ${formStateColor || contrastMediumColor}`,
+        borderRadius: borderRadiusSmall,
+        background: 'transparent',
+        textIndent: 0,
+        color: primaryColor,
+        transition: `${getTransition('background-color')}, ${getTransition('border-color')}, ${getTransition('color')}`, // for smooth transitions between e.g. disabled states
+        ...prefersColorSchemeDarkMediaQuery(theme, {
+          borderColor: formStateColorDark || contrastMediumColorDark,
+          color: primaryColorDark,
+        }),
+        gridArea: '1/1',
+        // TODO: move into getSlottedTextFieldTextareaSelectStyles()
+        font: textSmallStyle.font, // to override line-height
+        // TODO: move into getSlottedTextFieldTextareaSelectStyles()
+        padding: hasCounter
+          ? `12px ${formElementPaddingHorizontal} ${spacingStaticLarge}`
+          : `12px ${formElementPaddingHorizontal}`,
+      },
+      ...(hoverMediaQuery({
+        // with the media query the selector has higher priority and overrides disabled styles
+        ['textarea:not(:disabled):not(:focus):not([readonly]):hover,label:hover~.wrapper textarea:not(:disabled):not(:focus):not([readonly])']:
+          {
+            borderColor: formStateHoverColor || primaryColor,
+            ...prefersColorSchemeDarkMediaQuery(theme, {
+              borderColor: formStateHoverColorDark || primaryColorDark,
+            }),
+          },
+      }) as Styles),
+      // TODO: getFocusJssStyle() can't be re-used because focus style differs for form elements
+      ['textarea:focus']: {
```

Comment:
I think `[…]` is not needed? It's also possible to work with `&:focus` instead. Applies also to other selectors in this file.